components/kyma-environment-broker/internal/runtime/provider_test.go

```diff
@@ -0,0 +1,174 @@
+			given: given{
+				kymaVersion:                      "1.9.0",
+				managedRuntimeComponentsYAMLPath: path.Join("testdata", "managed-runtime-components.yaml"),
+			},
+			expectedRequestURL: "https://github.com/kyma-project/kyma/releases/download/1.9.0/kyma-installer-cluster.yaml",
+		},
+		{
+			name: "Provide on-demand Kyma version",
+			given: given{
+				kymaVersion:                      "master-ece6e5d9",
+				managedRuntimeComponentsYAMLPath: path.Join("testdata", "managed-runtime-components.yaml"),
+			},
+			expectedRequestURL: "https://storage.googleapis.com/kyma-development-artifacts/master-ece6e5d9/kyma-installer-cluster.yaml",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			installerYAML := readKymaInstallerClusterYAMLFromFile(t)
+			fakeHTTPClient := newTestClient(t, installerYAML, http.StatusOK)
+
+			listProvider := runtime.NewComponentsListProvider(fakeHTTPClient, tc.given.kymaVersion, tc.given.managedRuntimeComponentsYAMLPath)
+			expManagedComponents := readManagedComponentsFromFile(t, tc.given.managedRuntimeComponentsYAMLPath)
+
+			// when
+			allComponents, err := listProvider.AllComponents()
+
+			// then
+			require.NoError(t, err)
+			assert.NotNil(t, allComponents)
+
+			assert.Equal(t, tc.expectedRequestURL, fakeHTTPClient.RequestURL)
+			assertManagedComponentsAtTheEndOfList(t, allComponents, expManagedComponents)
+		})
+	}
+}
+
+func TestRuntimeComponentProviderGetFailures(t *testing.T) {
+	type given struct {
+		kymaVersion                      string
+		managedRuntimeComponentsYAMLPath string
+		httpErrMessage                   string
+	}
+	tests := []struct {
+		name          string
+		given         given
+		expErrMessage string
+	}{
+		{
+			name: "Provide release version not found",
+			given: given{
+				kymaVersion:                      "111.000.111",
+				managedRuntimeComponentsYAMLPath: path.Join("testdata", "managed-runtime-components.yaml"),
+				httpErrMessage:                   "Not Found",
+			},
+			expErrMessage: "while getting open source Kyma components list: got unexpected status code, want 200, got 404, url: https://github.com/kyma-project/kyma/releases/download/111.000.111/kyma-installer-cluster.yaml, body: Not Found",
+		},
+		{
+			name: "Provide on-demand version not found",
+			given: given{
+				kymaVersion:                      "master-123123",
+				managedRuntimeComponentsYAMLPath: path.Join("testdata", "managed-runtime-components.yaml"),
+				httpErrMessage:                   "Not Found",
+			},
+			expErrMessage: "while getting open source Kyma components list: got unexpected status code, want 200, got 404, url: https://storage.googleapis.com/kyma-development-artifacts/master-123123/kyma-installer-cluster.yaml, body: Not Found",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			fakeHTTPClient := newTestClient(t, tc.given.httpErrMessage, http.StatusNotFound)
```

Comment:
why didn't you use "net/http/httptest" package?

Reply:
the httptest gives you only http server. To use that you will need to inject the URL to provider and in current impl it is a provider responsibility to do that, so I can also assert that 
```go
assert.Equal(t, tc.expectedRequestURL, fakeHTTPClient.RequestURL)
```

So better is to use client and assert real request data